src/Handler/StreamHandler.php

```diff
@@ -63,8 +63,6 @@ public function __invoke(RequestInterface $request, array $options): PromiseInte
                 || \strpos($message, "connection attempt failed")
             ) {
                 $e = new ConnectException($e->getMessage(), $request, $e);
-            } else {
-                $e = RequestException::wrapException($request, $e);
```

Comment:
Not sure what kind of exception can actually occur at this point.

Reply:
Why do we have the if statement here then? Surely it can be entire removed, with lines 59 to 68 (original line numbers), simply replaced with the single line `$e = new ConnectException($e->getMessage(), $request, $e);`.

Reply:
The exceptions being tested in `testVerifiesVerifyIsValidIfPath()` and `testVerifiesCertIfValidPath()` occur here. 

Perhaps they should be wrapped in a `TransferException`?

Reply:
Or perhaps they should be invalid argument exceptions like this: https://github.com/guzzle/guzzle/blob/master/tests/Handler/CurlFactoryTest.php#L115

Reply:
I've made these changes in dc1591c - I can revert if you think it's incorrect.

Reply:
Please unresolve if this isn't correct